back-end/routes/users.js

```diff
@@ -1,55 +1,169 @@
+require('dotenv').config();
+
 var express = require('express');
 var router = express.Router();
+
 var bodyParser = require('body-parser');
-const { pool } = require('../config')
+const { pool } = require('../config');
+const {check, validationResult} = require('express-validator');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const expjwt = require('express-jwt');
 
-const makeDbQueryAndReturnResults = (queryString, res) => {
-  pool.query(queryString, (error, results) => {
-    if (!results || !results.rows || results.rows[0] === undefined || error) {
-      return res.status(500).send('An error occurred');
-    }
-    return res.status(200).send(results.rows);
-  })
-}
+const PASSWORD_MIN_LENGTH = 8;
 
-const makeDbQueryAndReturnStatus = (queryString, res) => {
+const makeDbQueryAndReturnResults = (queryString, res) => {
   pool.query(queryString, (error, results) => {
     if (error) {
+      return res.status(500).send(error);
+    }
+    else if (!results || !results.rows || results.rows[0] === undefined) {
       return res.status(500).send('An error occurred');
     }
-    return res.status(200);
+    return res.status(200).send(results.rows);
   })
-}
+};
 
-/* GET all users */
-router.get('/', function(req, res, next) {
-  const getAllUsersQuery = 'SELECT * FROM users';
-  makeDbQueryAndReturnResults(getAllUsersQuery, res);
-});
+// async function getRowFromDb(queryString) {
```

Comment:
I left this in as a reference, because it might be useful for later development...